synthesizer/process/src/stack/call/mod.rs

```diff
@@ -257,8 +268,66 @@ impl<N: Network> CallTrait<N> for Call<N> {
                         // Push the request onto the call stack.
                         call_stack.push(request.clone())?;
 
-                        // Execute the request.
-                        let response = substack.execute_function::<A, R>(call_stack, console_caller, rng)?;
+                        // Evaluate the request.
+                        let response = substack.evaluate_function::<A>(call_stack, console_caller)?;
+
+                        // Return the request and response.
+                        (request, response)
+                    }
+                    CallStack::CheckDeployment(_, private_key, ..) => {
+                        // Compute the request.
+                        let request = Request::sign(
+                            &private_key,
+                            *substack.program_id(),
+                            *function.name(),
+                            inputs.iter(),
+                            &function.input_types(),
+                            rng,
+                        )?;
+
+                        // Compute the address.
+                        let address = Address::try_from(&private_key)?;
+                        // Sample dummy outputs
+                        let outputs = function
+                            .outputs()
+                            .iter()
+                            .map(|output| match output.value_type() {
+                                ExternalRecord(locator) => {
+                                    // Retrieve the external stack.
+                                    let stack = substack.get_external_stack(locator.program_id())?;
+                                    // Sample the output.
+                                    stack.sample_value(&address, &Record(*locator.resource()), rng)
+                                }
```

Comment:
Why not move this case directly into the impl of `fn sample_value`?

Reply:
Smart a9f348306